ios/RNScreenshotDetector/RNScreenshotDetector.m

```diff
@@ -6,40 +6,212 @@
-                                                      [self screenshotDetected:notification];
-                                                  }];
+    screenshotObserver = [[NSNotificationCenter defaultCenter] 
+        addObserverForName:UIApplicationUserDidTakeScreenshotNotification
+        object:nil
+        queue:mainQueue
+        usingBlock:^(NSNotification *notification) {
+            [self screenshotDetected:notification];
+        }];
+    
+    if (@available(iOS 11.0, *)) {
+        screenRecordingObserver = [[NSNotificationCenter defaultCenter]
+            addObserverForName:UIScreenCapturedDidChangeNotification
+            object:nil
+            queue:mainQueue
+            usingBlock:^(NSNotification *notification) {
+                [self screenRecordingChanged:notification];
+            }];
+    }
 }
 
 - (void)stopObserving {
-    if (observer != nil) {
-        [[NSNotificationCenter defaultCenter] removeObserver:observer];
+    if (screenshotObserver != nil) {
+        [[NSNotificationCenter defaultCenter] removeObserver:screenshotObserver];
+        screenshotObserver = nil;
+    }
+    
+    if (screenRecordingObserver != nil) {
+        [[NSNotificationCenter defaultCenter] removeObserver:screenRecordingObserver];
+        screenRecordingObserver = nil;
     }
 }
 
 - (void)screenshotDetected:(NSNotification *)notification {
-    if (observer != nil) {
+    if (screenshotObserver != nil) {
         [self sendEventWithName:@"ScreenshotTaken" body:@{}];
     }
 }
 
-@end
+- (void)screenRecordingChanged:(NSNotification *)notification {
+    if (screenRecordingObserver != nil) {
+        BOOL isRecording = NO;
+        if (@available(iOS 11.0, *)) {
+            isRecording = [UIScreen mainScreen].isCaptured;
+        }
+        
+        [self sendEventWithName:@"ScreenRecordingChanged" body:@{@"isRecording": @(isRecording)}];
+        
+        if (isProtectionEnabled) {
+            if (isRecording) {
+                [self showSecurityOverlay];
+            } else {
+                [self hideSecurityOverlay];
+            }
+        }
+    }
+}
+
+RCT_EXPORT_METHOD(disableScreenshots) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        isProtectionEnabled = YES;
+        [self enableTrueScreenshotPrevention];
+        
+        if (@available(iOS 11.0, *)) {
+            BOOL currentlyRecording = [UIScreen mainScreen].isCaptured;
+            if (currentlyRecording) {
+                [self showSecurityOverlay];
+            }
+        }
+    });
+}
+
+RCT_EXPORT_METHOD(enableScreenshots) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        isProtectionEnabled = NO;
+        [self disableTrueScreenshotPrevention];
+        [self hideSecurityOverlay];
+    });
+}
+
+RCT_EXPORT_METHOD(isScreenRecording:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    if (@available(iOS 11.0, *)) {
```

Comment:
nit: we have minimum 16 version for exodus, so feel free to omit 
<img width="744" alt="image" src="https://github.com/user-attachments/assets/7ff9e8c3-d3de-40c7-bdbc-1ad4f0d0422a" />